src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTraitTest.php

```diff
@@ -0,0 +1,388 @@
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBag;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Security\Core\User\User;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+
+/**
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ControllerTraitTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGenerateUrl()
+    {
+        $router = $this->getMock(RouterInterface::class);
+        $router->expects($this->once())->method('generate')->willReturn('/foo');
+
+        $controller = new TestTrait();
+        $controller->setRouter($router);
+
+        $this->assertEquals('/foo', $controller->generateUrl('foo'));
+    }
+
+    public function testForward()
+    {
+        $request = Request::create('/');
+        $request->setLocale('fr');
+        $request->setRequestFormat('xml');
+
+        $requestStack = new RequestStack();
+        $requestStack->push($request);
+
+        $kernel = $this->getMock(HttpKernelInterface::class);
+        $kernel->expects($this->once())->method('handle')->will($this->returnCallback(function (Request $request) {
+            return new Response($request->getRequestFormat().'--'.$request->getLocale());
+        }));
+
+        $controller = new TestTrait();
+        $controller->setRequestStack($requestStack);
+        $controller->setHttpKernel($kernel);
+
+        $response = $controller->forward('a_controller');
+        $this->assertEquals('xml--fr', $response->getContent());
+    }
+
+    public function testRedirect()
+    {
+        $controller = new TestTrait();
+        $response = $controller->redirect('http://dunglas.fr', 301);
```

Comment:
`example.com` and `example.org` are reserved for documentation and examples: http://www.iana.org/domains/reserved  They are a global standard. They are guaranteed to work forever and it will always be safe to use them. No other domain (not even `symfony.com` itself) can guarantee that.